Userland/Utilities/tail.cpp

```diff
@@ -77,19 +51,87 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     TRY(Core::System::pledge("stdio rpath"));
 
     bool follow = false;
-    int line_count = DEFAULT_LINE_COUNT;
-    char const* file = nullptr;
+    size_t wanted_line_count = DEFAULT_LINE_COUNT;
+    StringView file;
 
     Core::ArgsParser args_parser;
     args_parser.set_general_help("Print the end ('tail') of a file.");
     args_parser.add_option(follow, "Output data as it is written to the file", "follow", 'f');
-    args_parser.add_option(line_count, "Fetch the specified number of lines", "lines", 'n', "number");
-    args_parser.add_positional_argument(file, "File path", "file");
+    args_parser.add_option(wanted_line_count, "Fetch the specified number of lines", "lines", 'n', "number");
+    args_parser.add_positional_argument(file, "File path", "file", Core::ArgsParser::Required::No);
     args_parser.parse(arguments);
 
-    auto f = TRY(Core::File::open(file, Core::OpenMode::ReadOnly));
-    TRY(Core::System::pledge("stdio"));
+    auto f = TRY(Core::Stream::File::open_file_or_standard_stream(file, Core::Stream::OpenMode::Read));
+    if (!follow)
+        TRY(Core::System::pledge("stdio"));
 
-    auto pos = find_seek_pos(*f, line_count);
-    return tail_from_pos(*f, pos, follow);
+    // f->tell() returns an error when the file is not seekable
+    auto potential_error = f->tell();
+    if (potential_error.is_error()) {
```

Comment:
```suggestion
    auto file_is_seekable = !f->tell().is_error();
    if (!file_is_seekable) {
```

If you think that this is reasonable (or a variation thereof).